Source/VCGeneration/VCGen.cs

```diff
@@ -28,6 +28,25 @@ public VCGen(Program program, string /*?*/ logFilePath, bool appendLogFile, List
       this.logFilePath = logFilePath;
     }
 
+    public static Graph<Block> BlocksToDag(List<Block> blocks)
```

Comment:
It is a great service to implement a method for creating a Graph from blocks.  But if the list of blocks has cycles, the output would not be a DAG.  This should not matter for this method whose only job is to create a Graph.  Please rename the method.

Also, there are places in the Boogie code where the same computation is reimplemented. It would be great to refactor the code so that all those places target this one method that you have introduced.  See, for example, the  method ComputeLiveVariables in DeadVarElim.cs.

Reply:
You might also consider splitting your PR into two PRs, one for the block->graph business and one for the splitting business.

Reply:
I thought the PR that introduced this general method already landed. Why is this method showing up as a diff?

Reply:
@typerSniper : Could you please respond to my question about BlocksToDAG so we can proceed to landing this PR?